src/mark.js

```diff
@@ -340,3 +340,14 @@ export function isTemporal(values) {
     return value instanceof Date;
   }
 }
+
+function isNumeric(values) {
+  for (const value of values) {
+    if (value == null) continue;
+    return !isNaN(+value);
+  }
+}
```

Comment:
This is a slightly different assertion than what we do elsewhere, e.g.:

https://github.com/observablehq/plot/blob/b1885ad65ff5351485260ec72acd12833969dbbb/src/scales.js#L104-L108

I think what we’re saying here is whether the values are “weakly” numeric, i.e., whether they can be treated as numbers (in this context). So renaming this to isWeaklyNumeric is probably good.

Also, I think we want to treat strict NaN as numeric in this context, since the goal here is to detect things that are _not_ numbers (such as strings), not to check that the numbers themselves are invalid. So, one possibility:

```suggestion
function isWeaklyNumeric(values) {
  for (const value of values) {
    if (value == null) continue;
    if (typeof value === "number") return true; // note: includes NaN!
    return !isNaN(+value);
  }
}
```